File: library/src/main/java/org/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairWriter.java
package org.crispy_fish.datatype.underscore_pairs;

import com.google.common.base.Preconditions;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

import static org.crispy_fish.datatype.underscore_pairs.Constants.UNDERSCORE;

public class SimpleStringUnderscorePairWriter implements UnderscorePairWriter<String> {

    public String pair(@Nonnull String key, @Nullable String value) {
        Preconditions.checkNotNull(key, "key must not be null");

        if (value == null) {
            value = "";
        }

        return UNDERSCORE + key + UNDERSCORE + value;
    }
}

File: library/src/main/kotlin/tech/coner/crispyfish/datatype/underscorepairs/SimpleStringUnderscorePairWriter.kt
package tech.coner.crispyfish.datatype.underscorepairs

class SimpleStringUnderscorePairWriter : UnderscorePairWriter<String> {

    override fun pair(key: String, value: String?): String {
        val writtenValue = value ?: ""
        return "_${key}_$writtenValue"
    }
}
